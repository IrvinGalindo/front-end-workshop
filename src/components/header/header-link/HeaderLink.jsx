import React from 'react';
import PropTypes from 'prop-types';

const HeaderLink = ({ onClick, label }) => (
  <li className="yta_Header-menu-list-item">
    <button type="button" onClick={onClick} className="yta_Link-underline--primary">
      {label}
    </button>
  </li>
);

HeaderLink.propTypes = {
  onClick: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
};

export default HeaderLink;
