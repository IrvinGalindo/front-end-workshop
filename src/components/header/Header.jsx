import React from 'react';

import './header.scss';
import HeaderLink from './header-link';

class Header extends React.Component {
  state = {
    loggedIn: false,
  };

  handleLogIn = () => {
    this.setState({ loggedIn: true });
  };

  handleLogOut = () => {
    this.setState({ loggedIn: false });
  };

  getPrivateLinks = () => (
    <React.Fragment>
      <HeaderLink onClick={() => console.log('adoptar')} label="Adoptar" />
      <HeaderLink onClick={() => console.log('rescatar')} label="Dar en adopción" />
      <HeaderLink onClick={this.handleLogOut} label="Cerrar sesión" />
    </React.Fragment>
  );

  getPublicLinks = () => <HeaderLink onClick={this.handleLogIn} label="Iniciar sesión" />;

  render() {
    const { loggedIn } = this.state;
    const links = loggedIn ? this.getPrivateLinks() : this.getPublicLinks();

    return (
      <header className="yta_Header">
        <h1 className="yta_Header-title">
          <span>Yo te</span> adopto
        </h1>
        <nav className="yta_Header-menu">
          <h2 className="yta_u-onlyRead">Menú de Navegación</h2>
          <ul className="yta_Header-menu-list">{links}</ul>
        </nav>
      </header>
    );
  }
}

export default Header;
