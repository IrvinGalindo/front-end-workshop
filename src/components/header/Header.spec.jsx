import React from 'react';
import { mount } from 'enzyme';

import Header from './Header';
import HeaderLink from './header-link';

describe('Header', () => {
  let header;

  beforeEach(() => {
    header = mount(<Header />);
  });

  it('should render three <HeaderLink />', () => {
    header.find('button').simulate('click');
    expect(header.find(HeaderLink).length).toBe(3);
    expect(header).toMatchSnapshot();
  });

  it('should render one <HeaderLink />', () => {
    header.find('button').simulate('click');
    header
      .find('button')
      .last()
      .simulate('click');
    expect(header.find(HeaderLink).length).toBe(1);
    expect(header).toMatchSnapshot();
  });
});
