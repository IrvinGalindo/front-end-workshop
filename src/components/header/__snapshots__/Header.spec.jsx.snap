// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render one <HeaderLink /> 1`] = `
<Header>
  <header
    className="yta_Header"
  >
    <h1
      className="yta_Header-title"
    >
      <span>
        Yo te
      </span>
       adopto
    </h1>
    <nav
      className="yta_Header-menu"
    >
      <h2
        className="yta_u-onlyRead"
      >
        Menú de Navegación
      </h2>
      <ul
        className="yta_Header-menu-list"
      >
        <HeaderLink
          label="Iniciar sesión"
          onClick={[Function]}
        >
          <li
            className="yta_Header-menu-list-item"
          >
            <button
              className="yta_Link-underline--primary"
              onClick={[Function]}
              type="button"
            >
              Iniciar sesión
            </button>
          </li>
        </HeaderLink>
      </ul>
    </nav>
  </header>
</Header>
`;

exports[`Header should render three <HeaderLink /> 1`] = `
<Header>
  <header
    className="yta_Header"
  >
    <h1
      className="yta_Header-title"
    >
      <span>
        Yo te
      </span>
       adopto
    </h1>
    <nav
      className="yta_Header-menu"
    >
      <h2
        className="yta_u-onlyRead"
      >
        Menú de Navegación
      </h2>
      <ul
        className="yta_Header-menu-list"
      >
        <HeaderLink
          label="Adoptar"
          onClick={[Function]}
        >
          <li
            className="yta_Header-menu-list-item"
          >
            <button
              className="yta_Link-underline--primary"
              onClick={[Function]}
              type="button"
            >
              Adoptar
            </button>
          </li>
        </HeaderLink>
        <HeaderLink
          label="Dar en adopción"
          onClick={[Function]}
        >
          <li
            className="yta_Header-menu-list-item"
          >
            <button
              className="yta_Link-underline--primary"
              onClick={[Function]}
              type="button"
            >
              Dar en adopción
            </button>
          </li>
        </HeaderLink>
        <HeaderLink
          label="Cerrar sesión"
          onClick={[Function]}
        >
          <li
            className="yta_Header-menu-list-item"
          >
            <button
              className="yta_Link-underline--primary"
              onClick={[Function]}
              type="button"
            >
              Cerrar sesión
            </button>
          </li>
        </HeaderLink>
      </ul>
    </nav>
  </header>
</Header>
`;
